src/sentry/integrations/vercel/webhook.py

```diff
@@ -191,22 +191,24 @@ def post(self, request: Request) -> Response | None:
                 payload = request.data["payload"]
                 external_id = self.parse_new_external_id(request)
                 scope.set_tag("vercel_webhook.type", "new")
+                new_webhook = True
 
                 if event_type == "integration-configuration.removed":
                     configuration_id = payload["configuration"]["id"]
-                    return self._delete(external_id, configuration_id, request)
+                    return self._delete(external_id, configuration_id, request, new_webhook)
                 if event_type == "deployment.created":
-                    return self._deployment_created(external_id, request)
+                    return self._deployment_created(external_id, request, new_webhook)
             except Exception:
                 external_id = self.parse_old_external_id(request)
                 scope.set_tag("vercel_webhook.type", "old")
+                new_webhook = False
```

Comment:
Is there any other way to check for this other than relying on an exception here? I'm a little confused about how this branch works, at what point is the exception thrown?

Reply:
I'm pretty sure the Exception is coming from `parse_new_eternal_id`. the new payload has `team` while the legacy version has `teamId` (or user/userId) so if it unable to find that key, it throws the Exception and it tries the old method

Reply:
Ok, I think we should rewrite this to be a little clearer. The code in these branches is pretty similar. Something roughly like
```
try:
    external_id = self.parse_new_external_id(request)
    new_webhook = True
    tag_type_string = 'new'
except <Better exception than Exception, we should be able to be specific. AttributeError?>:
    external_id = self.parse_old_external_id(request)
    new_webhook = False
    tag_type_string = 'old'

scope.set_tag("vercel_webhook.type", tag_type_string)

if event_type == "integration-configuration.removed":
    configuration_id = request.data["payload"]["configuration"]["id"]
    return self._delete(external_id, configuration_id, request, new_webhook)
if event_type == "deployment.created":
    return self._deployment_created(external_id, request, new_webhook)
```

Reply:
agreed - thanks for the suggestion